### PR TITLE
Adjust paths in extensions-sdk & publish composables types

### LINKED ISSUE #18178: upgrading to version 9.25.0
### Describe the Bug

@directus/utils and @directus/composables do not build/deploy with *.d.ts files.
image

@directus/extensions-sdk/cli.js is trying to run a file that doesn't exist.
image

### To Reproduce

- update extension to directus version > 9.25.0
- check exports from @directus/extensions-sdk package
- try to build it

### Hosting Strategy

Self-Hosted (Custom)

### PR DESCRIPTION
**Pull request by paescuj**
Fixes #18178

### CODE REVIEW

**Review by nickrum — APPROVED** [review]
LGTM code-wise! Haven't tested it, though.

File: packages/composables/package.json [hunk 1]
```diff
@@ -22,8 +22,7 @@
 	},
 	"main": "dist/index.js",
 	"files": [
-		"dist",
-		"!**/*.d.ts?(.map)"
+		"dist"
```
Comment (azrikahar) [inline packages/composables/package.json:25]: Do we need to add `!**/*.test.js` similar to the entry in api:

https://github.com/directus/directus/blob/c301cfbc82d1ce3ba8dad56800552c4899895b6e/api/package.json#L62

to ignore the built `use-items.test.js` test file? or maybe even another additional `!**/*.test.d.ts` if I'm not mistaken.

**Comment by paescuj** [comment]
Good eyes @azrikahar 👀😄 Works for you otherwise?

**Review by azrikahar — APPROVED** [review]
extensions-sdk fix LGTM and tested locally 👍 also added a unit test which does seem to fail pre-fix (to hopefully catch this if it ever happens again), and passes after this PR.

aside from the little `files` entry question for composables `package.json`, LGTM!

File: packages/composables/package.json (hunk 1 above)
Comment (azrikahar) [reply packages/composables/package.json:25]: let's do this separately if deemed necessary down the line.

Comment (azrikahar) [reply packages/composables/package.json:25]: Update: this is now addressed in #18211!

**Pull request merged**